#ifndef __PROTOBUF_MANAGER_HPP__
#define __PROTOBUF_MANAGER_HPP__

#include "servercommon/servercommon.h"
#include "google/protobuf/message.h"

enum PROTO_TYPE
{
	PROTO_ROLE_BATTLE_DATA,
	PROTO_PLATFORM_ROLE_DATA,
	PROTO_BATTLE_CHARACTER_DATA,
	PROTO_CLOUD_ARENA_FIGHT_RECORD_LIST_DATA,
	PROTO_ROLE_PET_INFO_DATA,
	PROTO_ROLE_PARTNER_INFO_DATA,
	PROTO_ROLE_MIRAGETERRITORY_INFO_DATA,
	PROTO_WORLD_ARENA_BASE_DATA,
	PROTO_WORLD_ARENA_USER_DATA,
	PROTO_WORLD_ARENA_SCHEDULE_DATA,
	PROTO_WORLD_ARENA_USER_SIGNUP_DATA,
	PROTO_WORLD_ARENA_USER_SCHEDULE_DATA,
	PROTO_WORLD_ARENA_USER_SCORE_DATA,
	PROTO_UTA_CHAMPION_RECORD_DATA,
	PROTO_UTA_CHAMPION_USER_TEAM_INFO_DATA,
	PROTO_UTA_TEAM_DATA,
	PROTO_UTA_TEAM_MEMBER_DATA,
	PROTO_UTA_GUESS_USER_INFO_DATA,
	PROTO_UTA_GUESS_TEAM_INFO_DATA,
	PROTO_UTA_LAST_SEASON_MAIN_OVERVIEW_DATA,
	PROTO_TYPE_MAX,
};

class ProtobufMgr
{
public:
	static ProtobufMgr* Instance();
	google::protobuf::Message* GetProtoMessage(int type);

private:
	google::protobuf::Message* m_proto_list[PROTO_TYPE_MAX];
	int m_used_times[PROTO_TYPE_MAX];

	INSTANCE_FORBID_COPY_AND_EXTERIAL_CREATE(ProtobufMgr);
};


#endif